homeassistant/components/shelly/device_trigger.py

```diff
@@ -0,0 +1,110 @@
+"""Provides device triggers for Shelly."""
+from typing import List
+
+import voluptuous as vol
+
+from homeassistant.components.automation import AutomationActionType
+from homeassistant.components.device_automation import TRIGGER_BASE_SCHEMA
+from homeassistant.components.device_automation.exceptions import (
+    InvalidDeviceAutomationConfig,
+)
+from homeassistant.components.homeassistant.triggers import event as event_trigger
+from homeassistant.const import (
+    ATTR_DEVICE_ID,
+    CONF_DEVICE_ID,
+    CONF_DOMAIN,
+    CONF_EVENT,
+    CONF_PLATFORM,
+    CONF_TYPE,
+)
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant
+from homeassistant.helpers.typing import ConfigType
+
+from .const import (
+    ATTR_CHANNEL,
+    ATTR_CLICK_TYPE,
+    CONF_SUBTYPE,
+    DOMAIN,
+    EVENT_SHELLY_CLICK,
+    SUPPORTED_TRIGGER_TYPES,
+    TRIGGER_SUBTYPES,
+)
+from .utils import get_device_wrapper, get_input_triggers
+
+TRIGGER_SCHEMA = TRIGGER_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_TYPE): vol.In(SUPPORTED_TRIGGER_TYPES),
+        vol.Required(CONF_SUBTYPE): vol.In(TRIGGER_SUBTYPES.keys()),
```

Comment:
Keys is default iterator.

```suggestion
        vol.Required(CONF_SUBTYPE): vol.In(TRIGGER_SUBTYPES),
```

Reply:
Fixed